engine/combat/combat_controller.py

```diff
@@ -41,10 +42,17 @@ def execute_combat(self: Self) -> bool:
         if combat_manager.encounter_done is True:
             return True
 
+        # if we are in the starting zone, just execute a non-timed basic attack unless
+        # we have already set the special action for the second encounter
+        # TODO(eein): Move these custom settings over to custom controllers after
+        # refactoring the controllers to use a base class and split methods.
+        # have it check if a state is valid, and set set a custom controller and
+        # execute on those actions, or use `self` to execute the default.
+        self._handle_starting_zone()
         # if some dialog is on the screen - make it go away
         if new_dialog_manager.dialog_open:
             self.ctrl.confirm()
-            self._handle_alternate_encounters()
+            self.second_encounter = True
```

Comment:
This seems like it wouldn't follow. There are other fights later where there is dialog during the fight, isn't there? So the `second_encounter` member variable seems misnamed.

Reply:
This will be moved later.